apps/passwd.c

```diff
@@ -556,7 +556,7 @@ static char *shacrypt(const char *passwd, const char *magic, const char *salt)
     OPENSSL_strlcat(out_buf, "$", sizeof out_buf);
     if (rounds_custom) {
         char tmp_buf[7 + 9 + 1]; /* "rounds=999999999" */
```

Comment:
max. unsigned long is 4294967295, 10 digits if long=32 bit
and 18446744073709551615, thus 20 digits if long=64 bit.